include/Stack.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+#include <Vector.h>
+#include <stdexcept>
+
+class Stack : private Vector
```

Comment:
`Stack` shouldn't inherit from `Vector`. Logically they don't have the relationship of "father and son". 
Instead `Stack` should hold `Vector` member attr.